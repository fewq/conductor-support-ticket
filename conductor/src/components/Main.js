import React, { Component } from "react";

export default class Main extends Component {
  render() {
    return (
      <div>
        <h1>Main component, {this.props.name}</h1>
        <h1>
          Click <a href="/restricted"> here</a> to access restricted area
        </h1>
      </div>
    );
  }
}
