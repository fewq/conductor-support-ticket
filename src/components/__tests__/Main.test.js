import React from 'react'
import Main from "../Main"
import { render, cleanup } from 'react-testing-library'
import renderer from 'react-test-renderer';
import {toBeInTheDocument, toHaveClass} from 'jest-dom'

expect.extend({toBeInTheDocument, toHaveClass})

afterEach(cleanup)

var mockAuth = {
    isAuthenticated: () => true
  };

test('renders greeting text', () => {
    const { getByText } = render(<Main auth={mockAuth} />)
    let greetingText = getByText(/Hey/);
    expect(greetingText).toBeInTheDocument();
  })

  test('snapshot testing', () => {
    const tree = renderer
    .create(<Main auth={mockAuth} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  