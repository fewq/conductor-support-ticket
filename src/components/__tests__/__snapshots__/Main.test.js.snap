// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot testing 1`] = `
<div>
  <div>
    <p>
      Hey, 
      , how can we assist you today?
    </p>
    <p>
      Click 
      <a
        href="/restricted"
      >
         here
      </a>
       to access restricted area
    </p>
  </div>
</div>
`;
